analysis.go

```diff
@@ -102,55 +106,57 @@ func checkProjectTask(checkFactory getProjectCheckFactory, project, basepath str
 	}
 }
 
-// getDumpedResourceAsStruct retrieves the requested resource from the dump directory and
-// decodes the JSON into the provided interface
-func getDumpedResource(path []string, dest interface{}) error {
-	filePath := filepath.Join(path...)
-	contents, err := ioutil.ReadFile(filePath)
+// Takes an array of strings describing the path to a JSON file
+// which will be parsed and loaded into the supplied interface
+type DumpedJSONResourceFactory func([]string, interface{}) error
+
+// Returns a factory which will load resource from a given basepath. The factory parses the file
+// contents as JSON and loads it into the provided dest interface
+func getDumpedJSONResourceFactory(basepath []string) DumpedJSONResourceFactory {
+	return func(path []string, dest interface{}) error {
+		file := filepath.Join(append(basepath, path...)...)
+		contents, err := ioutil.ReadFile(file)
+
```

Comment:
Any reason to read the whole thing into mem? Would opening the file and passing the file object that is a reader to  the decoder be an option. Not sure what the encoder does underneath there. Likely has to read the whole thing cause json can't be partially parsed. So there may not be a benefit